store/task.go

```diff
@@ -745,9 +755,13 @@ func (s *Store) patchTaskStatusImpl(ctx context.Context, tx *sql.Tx, patch *api.
 			return nil, FormatError(err)
 		}
 		taskPatchedRaw = &taskRaw
+		found = true
 	}
-	if err := row.Close(); err != nil {
-		return nil, err
+	if err := row.Err(); err != nil {
+		return nil, FormatError(err)
+	}
```

Comment:
Let's call `row.Close` after `row.Err`

Reply:
IMO, `defer row.Close()` already closes the row after `row.Err()`.

Reply:
The reason why we should call `row.Close` explicitly even if `row.Err() === nil` is that we use the same `tx` afterward so we should close this connection by calling `row.Close`.